api.js

```diff
@@ -715,6 +751,55 @@
     }
   };
 
+  var cachedRoutes = {
+    'get': {
+      '/user/find/(a-zA-Z0-9)': function(urlParts, params) {
+        // Use everything for the hash
+        return [urlParts, orderObject(params)];
+      }
+    },
+    'get-expires': {
+      '/user/find/(a-zA-Z0-9)': 300000 // 300s / 5m
+    }
+  };
+
+  // Returns an ordered version of it's input
+  function orderObject(source) {
+    var output = {};
+    Object.keys(source).sort().forEach(function(key) {
+      output[key] = source[key];
+    });
+    return output;
+  }
+
+  function qhash(input) {
+    switch(typeof input) {
+      case 'string':
+        // do nothing
+        break;
+      case 'object':
+        input = orderObject(input);
+      default:
+        input = JSON.stringify(input);
+        break;
+    }
+
+    if(crypto) {
+      return crypto.createHash('sha256').update(input).digest('base64');
+    }
+
+    // Fallback hash function
+    // http://stackoverflow.com/a/7616484/2928176
+    var hash = 0, i, chr, len;
```

Comment:
Because those will become VERY lengthy on hashing a string over a 1000 characters long (longer than the keys in localstorage are allowed to be, or cookies for the fallback)